merlin/study/batch.py

```diff
@@ -110,16 +111,19 @@ def get_node_count(default=1):
     return default
 
 
-def batch_worker_launch(spec, com, nodes=None, batch=None):
+def batch_worker_launch(spec: Dict,
+                        com: str,
+                        nodes: Union[str, Dict] = None,
+                        batch=None) -> str:
```

Comment:
Hmm - the docstring's untouched from main and as is only indicates the desired type, not accepted types the variable is expected to possibly hold when passed or during the function, which is what's needed (if datatype changes between passing and what's handled internally, I prefer two variables - one for better type enforcement, to make sure what's passed is valid, and one that's defined in while processed the passed var to whatever type is needed elsewhere).

Seeing it again, both the docstring and signature are wrong, this should be Optional because None is valid, the union is missing int, and Dict isn't valid. Mypy would catch those discrepancies when we bring it online.

I'll fix both.

Reply:
the nodes field in the header should be an int?

Reply:
It's needed as an int but at lines 142 and 147 `nodes` can be a None, int, or str. We can declare it only an int in the header/signature, but if/when we do type enforcement with mypy it would fail out at those lines.

I added a check to make sure that, at the point `nodes` is supposed to have been processed to an int, it is actually an int.